controllers/link.go

```diff
@@ -146,6 +151,55 @@ func GetLinkWithPassword(c *gin.Context) {
 	c.JSON(http.StatusOK, link)
 }
 
+func UpdateLink(c *gin.Context) {
+	var input UpdateLinkInput
+	if err := c.ShouldBindJSON(&input); err != nil {
+		c.JSON(http.StatusUnprocessableEntity, err.Error())
+		return
+	}
+
+	if input.Password == "" && input.Alias == "" {
+		c.JSON(http.StatusNoContent, http.StatusText(http.StatusNoContent))
+		return
+	}
+
+	user, _ := authentication.GetCurrentUserFromContext(c)
+	id, err := strconv.ParseUint(c.Param("id"), 10, 64)
+	if err != nil {
+		c.JSON(http.StatusNotFound, http.StatusText(http.StatusNotFound))
+		return
+	}
+
+	var link = new(models.Link)
+	database.DB.Where(&models.Link{
+		UserID: user.ID,
+		BaseModel: models.BaseModel{
+			ID: uint(id),
+		},
+	}).First(link)
+
+	if link.ID == 0 {
+		c.JSON(http.StatusNotFound, http.StatusText(http.StatusNotFound))
+		return
+	}
+
+	if input.Password != "" {
+		link.Password = input.Password
+	}
+
+	if input.Alias != "" {
+		if linkutils.IsDuplicateAlias(input.Alias) {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Duplicate Alias"})
```

Comment:
```suggestion
			c.JSON(http.StatusBadRequest, gin.H{"error": "This alias  is already used"})
```
Would be more informative, what do you think?